corehq/blobs/models.py

```diff
@@ -4,3 +4,17 @@
 class BlobMigrationState(models.Model):
     slug = models.CharField(max_length=20, unique=True)
     timestamp = models.DateTimeField(auto_now=True)
+
+
+class BlobExpiration(models.Model):
+    '''
+    This models records when temporary blobs should be deleted
+    '''
+    bucket = models.CharField(max_length=255)
+    identifier = models.CharField(max_length=255)
+    expires_on = models.DateTimeField(db_index=True)
+    created_on = models.DateTimeField(auto_now=True)
+    length = models.IntegerField()
+
+    # This is set to True when the blob associated has been successfully deleted
+    deleted = models.BooleanField(default=False, db_index=True)
```

Comment:
thinking about it more, we may want to actually update the blob expiration record when the same bucket/identifier is used rather than create a new one